From 6b1c41352c8be42fd2d3cb9c2f2be0415db4fe16 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Tue, 30 Dec 2025 16:02:58 +0100
Subject: [PATCH] Specify libgnome_games_support as dependency

Without this, meson does not wait util the compilation of
libgnome_games_support is finished, causing a build failure.
---
 meson.build     | 2 +-
 src/meson.build | 5 +++--
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/meson.build b/meson.build
index 87f877c..a7fd243 100644
--- a/meson.build
+++ b/meson.build
@@ -24,7 +24,7 @@ gtk_dependency = dependency('gtk4', version: '>= 4.16.0')
 adw_dependency = dependency('libadwaita-1', version: '>= 1.8')
 libgnome_games_support = subproject ('libgnome-games-support',
                                      default_options: 'superproject_name=gnome-2048')
-libgnome_games_support_dependency = libgnome_games_support.get_variable ('lggs_dependency')
+libgnome_games_support_dependency = libgnome_games_support.get_variable ('lggs')
 
 cargo = find_program('cargo', required: true)
 cargo_version = run_command(cargo, '--version', check: true).stdout().strip()
diff --git a/src/meson.build b/src/meson.build
index c1ee319..acaa7cd 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -40,6 +40,7 @@ custom_target(
   'gnome-2048',
   build_by_default: true,
   build_always_stale: true,
+  depends: [libgnome_games_support_dependency],
   input: [config_rs],
   output: meson.project_name(),
   console: true,
@@ -58,6 +59,6 @@ test(
   'cargo-test', cargo,
   env: cargo_env,
   args: ['test', '--no-fail-fast'] + cargo_options,
-  # depends: [libgnome_games_support_dependency],
+  depends: [libgnome_games_support_dependency],
   timeout: 180
-)
\ No newline at end of file
+)
-- 
GitLab

